sentinel2-processing/prepare_10m_sentinel_imagery.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+
+"""
+This is a Python version of a bash script written by Rebecca Clarke https://gist.github.com/rebclarke/b9e55a3558be3ec76248c78379b67366
+"""
+
+import os
+import glob
+import shutil
+from osgeo import gdal
+import subprocess
+
+
+# user definated variables
+TILE_INDEX = "nz-150k-tile-index.shp"
+MIN = "800"
+MAX = "3000"
+CWD = "C:/Temp/image-processing/test"
+SENTINEL_IMAGE_DIR = "R10m"
+GDAL_DIR = "C:/Program Files/QGIS 3.34.4/apps/Python39/Scripts"
+
+
+def make_directories(dir_name):
+    """
+    Create directors for the various processed images to be saved into.
+    """
+    path = os.path.join(CWD, dir_name)
+    if os.path.exists(path):
+        shutil.rmtree(path)
+    os.mkdir(path)
+
+
+# make output directories
+make_directories("output")
+make_directories("output/merged")
+make_directories("output/rescaled")
+make_directories("output/nearblack")
+make_directories("output/reprojected")
+make_directories("output/tiled")
+
+
+path = os.path.join(CWD, SENTINEL_IMAGE_DIR)
+rgb_files = glob.glob("R10m/*B0[2-4]_10m.jp2")
+
+blue = rgb_files[0]
+green = rgb_files[1]
+red = rgb_files[2]
+
+gdal_merge_path = os.path.join(GDAL_DIR, "gdal_merge.py")
+
+merged_output = os.path.join(CWD, "output/merged", "merged.tif")
+translate_output = os.path.join(CWD, "output/rescaled", "rescaled.tif")
+nearblack_output = os.path.join(CWD, "output/nearblack", "nearblack.tif")
+reprojected_output = os.path.join(CWD, "output/reprojected", "reprojected.tif")
+tiled_output_dir = os.path.join(CWD, "output/tiled")
+vrt_output = os.path.join(CWD, "output", "overall.vrt")
+
+# merge the three R,G,B images together to create a single RGB image
+print("Merging image tiles into one RGB image")
+subprocess.run(
+    [
+        "python.exe",
+        gdal_merge_path,
+        "-separate",
+        "-co",
+        "PHOTOMETRIC=RGB",
+        "-o",
+        merged_output,
+        red,
+        green,
+        blue,
+    ]
+)
+
+# rescale the imagery to 8bit unsigned
+print("Rescaling to  8bit unsigned")
+subprocess.run(
+    [
+        "gdal_translate",
+        "-ot",
+        "Byte",
+        "-of",
+        "GTiff",
+        "-scale",
+        MIN,
+        MAX,
+        "1",
+        "255",
```

Comment:
According to [the documentation](https://gdal.org/programs/gdal_translate.html#cmdoption-gdal_translate-scale), "If omitted the output range is 0 to 255". Do we actually want to use the output range 1 to 255 instead?